ueberdosis/tiptap
# Fix inline code formatting for text enclosed in backticks

## Please describe your changes

In the current implementation of the [@tiptap/extension-code](https://github.com/ueberdosis/tiptap/blob/39cf6979c49e953118bbbe4b894a1dc296128932/packages/extension-code/src/code.ts#L4), the regex pattern does not correctly format text enclosed in backticks when they are immediately preceded or followed by any character other than whitespace.

## How did you accomplish your changes

Updated the `inputRegex` and `pasteRegex` patterns to the following:

```ts
export const inputRegex = /(?<!`)`([^`]+)`(?!`)/;
export const pasteRegex = /(?<!`)`([^`]+)`(?!`)/g;
```

The regex uses negative lookbehind and lookahead assertions to ensure that only single backticks are considered for formatting to avoid conflicts with potential code block syntax or other uses of multiple backticks.

## How have you tested your changes

Tested in various scenarios to ensure that it correctly formats text as inline code blocks, and that it doesn't conflict with other extensions.

<img width="1125" alt="TipTap-LocalTest" src="https://github.com/ueberdosis/tiptap/assets/99911676/cdb6b076-4d71-4305-a508-dda793003f63">

## How can we verify your changes

1. Clone the repository and switch to the branch containing these changes.
2. Start the app via `npm run start` and navigate to any playground
3. Type text enclosed in backticks, with various surrounding characters.
4. You should observe the text within the backticks being correctly formatted as inline code.
5. Additionally, test with multiple backticks to ensure they don't interfere with the formatting.

## Remarks

This change enhances the flexibility of the inline code formatting.

## Checklist


## Related issues

#4467 
[Issue #338 in BlockNote](https://github.com/TypeCellOS/BlockNote/issues/338) - Original issue where the bug was identified and discussed.
